pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -459,6 +463,22 @@ func (r *ReconcileClusterDeployment) syncDeletedClusterDeployment(cd *hivev1.Clu
 		cdLog.WithField("jobName", installJob.Name).Info("install job deleted")
 	}
 
+	// Skips creation of uninstall job if PreserveOnDelete is true
+	if cd.Spec.PreserveOnDelete {
+		cdLog.Warn("skipping creation of uninstall job")
+		/*if !HasFinalizer(cd, hivev1.FinalizerDeprovision) {
+			return reconcile.Result{}, nil
+		} else {
+			cdLog.Warn(cd.Finalizers)
+			err = r.removeClusterDeploymentFinalizer(cd)
+			if err != nil {
+				cdLog.WithError(err).Error("error removing finalizer")
+			}
+			return reconcile.Result{}, err
+		}*/
+		return reconcile.Result{}, nil
```

Comment:
Just do the if HasFinalizer { remote it } here. As far as I can tell that should be all we need.